packages/resources-provider-plugin/src/index.ts

```diff
@@ -182,6 +184,60 @@ module.exports = (server: ResourceProviderApp): Plugin => {
     server.setPluginStatus(msg)
   }
 
+  /** process changes in config schema */
+  const cleanConfig = (options: ProviderSettings): ProviderSettings => {
+    server.debug(`Check / Clean loaded settings...`)
+
+    const defaultConfig: ProviderSettings = {
+      standard: {
+        routes: true,
+        waypoints: true,
+        notes: true,
+        regions: true,
+        charts: true
+      },
+      custom: []
+    }
+
+    // set / save defaults if no saved settings
+    if (!options?.standard) {
+      server.savePluginOptions(defaultConfig, () => {
+        server.debug(`Default configuration applied...`)
+      })
+      return defaultConfig
+    }
+
+    let isCleaned = false
+    // check / clean settings
+    if (!Array.isArray(options?.custom)) {
+      options.custom = []
+      isCleaned = true
+    }
+    if (typeof options.standard.charts === 'undefined') {
+      options.standard.charts = false
+      isCleaned = true
+    }
+    let idx = -1
+    for (let i = 0; i < options.custom.length; i++) {
+      if (options.custom[i].name === 'charts') {
+        idx = i
+      }
+    }
```

Comment:
Wouldn't [Array.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) be easier to understand? That's what this does, right?